src/ui_basic/panel.cc

```diff
@@ -797,6 +795,10 @@ void Panel::do_draw_inner(RenderTarget& dst) {
  * \param dst RenderTarget for the parent Panel
  */
 void Panel::do_draw(RenderTarget& dst) {
+	// TODO(Niektory): Are the asserts needed/correct?
```

Comment:
This is the point in the code where panel size matters, so shifting it here is good. Can 4k displays become higher than 3000 though?

Reply:
As in height? Typically no, but if you rotate them or stack two of them vertically then yes. With this feature you can easily stretch the window over multiple monitors :-)

Reply:
I meant both dimensions. How about setting it to 10000 or something? The purpose of the upper bound it to make sure there are no really crazy values, e.g. due to erroneous initialization.

Reply:
Can be 10000, sure. I don't really have an opinion on this. I just had to change it in this PR because it was too easy to trigger it by just resizing the window.